src/composition/supergraph/config/resolve/mod.rs

```diff
@@ -193,6 +192,39 @@ impl LazilyResolvedSupergraphConfig {
             Err(errors)
         }
     }
+
+    /// Fully resolves a [`LazilyResolvedSupergraphConfig`] into a [`FullyResolvedSubgraphs`]
+    /// by retrieving all the schemas as strings
+    pub async fn fully_resolve_subgraphs(
+        self,
+        introspect_subgraph_impl: &impl IntrospectSubgraph,
+        fetch_remote_subgraph_impl: &impl FetchRemoteSubgraph,
+    ) -> Result<FullyResolvedSubgraphs, Vec<ResolveSubgraphError>> {
+        let subgraphs = stream::iter(self.subgraphs.into_iter().map(
+            |(name, lazily_resolved_subgraph)| async {
+                let result = FullyResolvedSubgraph::fully_resolve(
+                    introspect_subgraph_impl,
+                    fetch_remote_subgraph_impl,
+                    lazily_resolved_subgraph,
+                    name.clone(),
+                )
+                .await?;
+                Ok((name, result))
+            },
+        ))
+        .buffer_unordered(50)
+        .collect::<Vec<Result<(String, FullyResolvedSubgraph), ResolveSubgraphError>>>()
+        .await;
+        let (subgraphs, errors): (
+            Vec<(String, FullyResolvedSubgraph)>,
+            Vec<ResolveSubgraphError>,
+        ) = subgraphs.into_iter().partition_result();
+        if errors.is_empty() {
+            Ok(subgraphs.into())
+        } else {
+            Err(errors)
+        }
+    }
```

Comment:
This is very similar to the method for doing this for the FullyResolvedSupergraphConfig but with enough difference to not want to refactor it into a single method

Reply:
I wonder if we want to look at the two, largely because resolving the federation version consistently would be useful. The current implementation here doesn't really take into account subgraphs, unless I missed something.